### PR TITLE
Harmonized stage code draft implementation

### LINKED ISSUE #48: Provide letter stage code mapping/linking to/from stage numbers (and URN stages)
I found a problem with pubid-iso, I can't update stage using number for Pubid::Iso::Identifier.
```
pubid = Pubid::Iso::Identifier.parse("ISO/IEC 17025:2005")
pubid.stage = "50.00"
```

`pubid.to_s` will return error because it's expecting one of stages abbreviation (like FDIS or CD)

It's ok, until I want to generate URN with stage number like "90.93", with current implementation it's impossible.

### PR DESCRIPTION
**Pull request by mico**
closes #48

### CODE REVIEW

**Comment by mico** [comment]
@ronaldtse Just a draft implementation to check if I'm on the right track.

File: lib/pubid/iso/harmonized_stage_code.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+module Pubid::Iso
+  class HarmonizedStageCode
+    attr_accessor :stage, :substage
+
+    STAGES = { PWI: "00.00",
```
Comment (ronaldtse) [inline lib/pubid/iso/harmonized_stage_code.rb:5]: Let's not have these "abbreviated stages" in this class. These codes only applies to ISO documents, they are not inherent to the HSC (harmonized stage codes). Let's move this outside of this class but keep in ISO.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: lib/pubid/iso/harmonized_stage_code.rb (hunk 1 above)
Comment (mico) [reply lib/pubid/iso/harmonized_stage_code.rb]: @ronaldtse I moved it to separate class "Stage". Could you check the implementation?

File: lib/pubid/iso/errors.rb [hunk 2]
```diff
@@ -2,5 +2,7 @@ module Pubid::Iso
   module Errors
     class ParseError < StandardError; end
     class PublishedIterationError < StandardError; end
+    class HarmonizedStageCodeNotValidError < StandardError; end
```
Comment (ronaldtse) [inline lib/pubid/iso/errors.rb:5]: `NotValid` => `Invalid`?

File: lib/pubid/iso/errors.rb [hunk 3]
```diff
@@ -2,5 +2,7 @@ module Pubid::Iso
...
+    class CodeNotValidError < StandardError; end
```
Comment (ronaldtse) [inline lib/pubid/iso/errors.rb:6]: `NotValid` => `Invalid`?

**Review by ronaldtse — CHANGES_REQUESTED** [review]
Almost there. Thanks @ni4 !

**Comment by ni4** [comment]
> Almost there. Thanks @ni4 !

I guess I was mentioned here by mistake :)

File: lib/pubid/iso/transformer.rb [hunk 4]
```diff
@@ -104,22 +104,23 @@ class Transformer < Parslet::Transform
 
     def self.convert_stage(code)
       russian_code = Renderer::Russian::STAGE.key(code.to_s)
```
[the remark this answers is not in the document]
Comment (mico) [reply lib/pubid/iso/transformer.rb:106]: @ronaldtse It's just a conversion of language-specific elements to generic ones, for example:
"ОПМС" become "FDIS"

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
(Sorry @ni4 !!)

Thanks @mico this is good to go!

**Pull request merged**